StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs

```diff
@@ -218,6 +263,11 @@ private void HandleEventDeclaration(SyntaxNodeAnalysisContext context)
 
         private void HandleEventFieldDeclaration(SyntaxNodeAnalysisContext context)
         {
+            if (context.GetDocumentationMode() != DocumentationMode.Diagnose)
+            {
+                return;
+            }
+
```

Comment:
Not without additional refactoring. Each of these methods is its own entry and exit point for analysis of a particular type of syntax node.